objects/src/notes/nullifier.rs

```diff
@@ -0,0 +1,105 @@
+use super::{Digest, Felt, Hasher, Note, Word, WORD_SIZE, ZERO};
+use crate::utils::serde::{
+    ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+};
+
+// NULLIFIER
+// ================================================================================================
+
+/// A note's nullifier.
+///
+/// A note's nullifier is computed as hash(serial_num, script_hash, input_hash, vault_hash).
+///
+/// This achieves the following properties:
+/// - Every note can be reduced to a single unique nullifier.
+/// - We cannot derive a note's hash from its nullifier.
```

Comment:
`We cannot derive a note's hash from its nullifier.` 

Shouldn't be the other way around important? That we cannot derive the note's nullifier from its hash? The hash is public knowledge, but the nullifier should be hidden to ensure private consumption, right?

Reply:
or both is important

Reply:
Both are important - I'll change this.